include/oxen/quic/gnutls_crypto.hpp

```diff
@@ -259,20 +260,125 @@ namespace oxen::quic
         }
     };
 
+    struct gtls_session_ticket
```

Comment:
I find this class a bit more complex than it needs to be: it has a sort of dual mode here where it's a copyable, movable object but then when an external application deals with it, it is expected to ignore that and keep it in a unique_ptr.

I took a stab at simplifying it to simply *always* force it to be used via unique_ptr, which allows dropping some of the pointer updating logic; I've pushed the commit to this PR.

That commit also makes `_key` const, documenting its suitability for use as the key for a map (addressing my earlier comment about the safety of the string_view key).